cluster-autoscaler/core/scale_down_test.go

```diff
@@ -176,6 +176,63 @@ func TestFindUnneededNodes(t *testing.T) {
 	assert.Equal(t, 0, len(sd.unremovableNodes))
 }
 
+func TestFindUnneededMoveToEmptyWithMinSizeReached(t *testing.T) {
+	provider := testprovider.NewTestCloudProvider(nil, nil)
+
+	// node group with empty node, but min size reached
+	provider.AddNodeGroup("ng1", 1, 10, 1)
+	n1 := BuildTestNode("n1", 1000, 10)
+	SetNodeReadyState(n1, true, time.Time{})
+	provider.AddNode("ng1", n1)
+
+	// node group with one underutilized node and node min size not yet reached
+	provider.AddNodeGroup("ng2", 1, 10, 2)
+	n2 := BuildTestNode("n1", 1000, 10)
```

Comment:
Shouldn't this node be called "n2"? Right now there are 2 nodes named "n1" and p1 pod is assigned to a non-existent "n2".

Reply:
(Test no longer passes when this is changed, BTW)